### PR TITLE
Support multiple ldap or AD servers

### PR DESCRIPTION
**Pull request by mayrstefan**
Support multiple ldap and AD hosts on the API level as requested in #25557

The implementation is not very sophisticated - if we can not connect to a server we go to the next one in the list. This pull requests only enables setting this on the API level. Multiple servers are not supported in the UI yet.

### CODE REVIEW

**Comment by mayrstefan** [comment]
I'm surprised that default-linux-amd64 does not build but all other architectures do. Flaky CI system?

Also i didn't find output for test_auth.py. Do we need any magic environment variables to enable this test?

**Comment by luthermonson** [comment]
@slickwarren any chance you know of a way to QA this with two ldap servers? i think this is done as far as code but we need to give it a once over with a QA pass so it's able to be merged.

**Comment by slickwarren** [comment]
we have both AD and azureAD LDAP setup, I think that would be sufficient to use in this case.

File: tests/validation/tests/v3_api/test_auth.py
Comment (slickwarren) [inline tests/validation/tests/v3_api/test_auth.py:613]: this looks like duplicated data, is this line correct? I would expect this to take a new variable that is a list itself or something of the sort, instead of the same variable 2x

File: pkg/auth/providers/common/ldap/ldap_util.go
Comment (slickwarren) [inline pkg/auth/providers/common/ldap/ldap_util.go:45]: will we have UI support for this as well?

File: tests/validation/tests/v3_api/test_auth.py
Comment (mayrstefan) [reply tests/validation/tests/v3_api/test_auth.py:613]: Yes, this line is intended like this: the API should accept a list of more than one server. And I did not want to test any error cases when that host is not reachable. I was not brave enough to replace the first list entry with a wrong servername because my local dapper ci tests don't execute that test. And I have no idea how to enable that.

File: pkg/auth/providers/common/ldap/ldap_util.go
Comment (mayrstefan) [reply pkg/auth/providers/common/ldap/ldap_util.go:45]: Yes, the old UI already has the necessary code to accept a comma separated list and split it into a list. For dashboard I had to re-add that feature. See [#5606](https://github.com/rancher/dashboard/pull/5606/files)

Comment (mayrstefan) [reply pkg/auth/providers/common/ldap/ldap_util.go:45]: For reference: 
https://github.com/rancher/ui/blob/543324df8cdb9472eee3950ad3e8b9b45401d749/lib/global-admin/addon/security/authentication/activedirectory/controller.js#L102-L111

**Review by slickwarren — APPROVED** [review]

**Review by luthermonson — APPROVED** [review]
We will likely merge this and pass to QA after v2.6.5 drops shortly. Thanks again @mayrstefan for the community PR!

**Review by jiaqiluo — APPROVED** [review]
LGTM

**Comment by mayrstefan** [comment]
v2.6.5 was released two weeks ago. Do we need anything to change?

**Comment by Jono-SUSE-Rancher** [comment]
@jiaqiluo - This looks good and has approvals from everyone, can we have this merged so it unblocks the associated Dashboard issue?

**Pull request merged**
